otherlibs/systhreads/st_stubs.c

```diff
@@ -70,59 +61,72 @@ struct caml_thread_descr {
 /* The infos on threads (allocated via caml_stat_alloc()) */
 
 struct caml_thread_struct {
+
   value descr;              /* The heap-allocated descriptor (root) */
-  struct caml_thread_struct * next;  /* Double linking of running threads */
+  struct caml_thread_struct * next; /* Doubly-linked list of running threads */
   struct caml_thread_struct * prev;
-#ifdef NATIVE_CODE
-  char * top_of_stack;      /* Top of stack for this thread (approx.) */
-  char * bottom_of_stack;   /* Saved value of Caml_state->bottom_of_stack */
-  uintnat last_retaddr;     /* Saved value of Caml_state->last_return_address */
-  value * gc_regs;          /* Saved value of Caml_state->gc_regs */
-  char * exception_pointer; /* Saved value of Caml_state->exception_pointer */
-  struct caml__roots_block * local_roots; /* Saved value of local_roots */
-  struct longjmp_buffer * exit_buf; /* For thread exit */
-#else
-  value * stack_low; /* The execution stack for this thread */
-  value * stack_high;
-  value * stack_threshold;
-  value * sp;        /* Saved value of Caml_state->extern_sp for this thread */
-  value * trapsp;    /* Saved value of Caml_state->trapsp for this thread */
-  /* Saved value of Caml_state->local_roots */
-  struct caml__roots_block * local_roots;
-  struct longjmp_buffer * external_raise; /* Saved Caml_state->external_raise */
+  int domain_id;      /* The id of the domain to which this thread belong to */
+  struct stack_info* current_stack;      /* saved Caml_state->current_stack */
+  struct c_stack_link* c_stack;          /* saved Caml_state->c_stack */
+  struct caml__roots_block *local_roots; /* saved value of local_roots */
+  struct longjmp_buffer *exit_buf;       /* For thread exit */
+  int backtrace_pos;           /* saved value of Caml_state->backtrace_pos */
+  code_t * backtrace_buffer;   /* saved value of Caml_state->backtrace_buffer */
+  value backtrace_last_exn;
+    /* saved value of Caml_state->backtrace_last_exn (root) */
+  value * gc_regs;           /* saved value of Caml_state->gc_regs */
+  value * gc_regs_buckets;   /* saved value of Caml_state->gc_regs_buckets */
+  value ** gc_regs_slot;     /* saved value of Caml_state->gc_regs_slot */
+  void * exn_handler;        /* saved value of Caml_state->exn_handler */
+
+#ifndef NATIVE_CODE
+  intnat trap_sp_off;      /* saved value of Caml_state->trap_sp_off */
+  intnat trap_barrier_off; /* saved value of Caml_state->trap_barrier_off */
+  /* saved value of Caml_state->external_raise */
+  struct caml_exception_context* external_raise;
```

Comment:
```suggestion
  struct caml_exception_context* external_raise;
    /* saved value of Caml_state->external_raise */
```

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/32e8f0f0a6055da5bf1330a69d62756f48ac214e